File: 백준/Silver/1213. 팰린드롬 만들기/팰린드롬 만들기.cc
#include <bits/stdc++.h>
using namespace std;

string s, ret;
int cnt[200], flag;
char mid;

int main(){
    cin >> s;
    for(char a : s) cnt[a]++;
    for (int i = 'Z'; i >= 'A'; i--){
        if (cnt[i]){
            if (cnt[i] & 1){ // 비트 연산자
                mid = char(i); flag++;
                cnt[i]--;
            }
            if (flag == 2) break;
            for (int j = 0; j < cnt[i]; j += 2){
                ret = char(i) + ret;
                ret += char(i);
            }
        }
    }    

    if(mid)ret.insert(ret.begin() + ret.size() / 2, mid);
    if (flag == 2) cout << "I'm Sorry Hansoo\n";
    else cout << ret << "\n";
}

File: 백준/Silver/1213. 팰린드롬 만들기/팰린드롬 만들기.java
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws Exception {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        String s = br.readLine();
        int[] a = new int[26];

        for (int i = 0; i < s.length(); i++) {
            int idx = s.charAt(i) - 'A';
            a[idx]++;
        }

        int oddCount = 0;
        int mid = 0;

        for (int i = 0; i < a.length; i++) {
            if (a[i] == 0) {
                continue;
            }

            if (a[i] % 2 == 1) {
                oddCount++;
                mid = i;
            }

            for (int j = 0; j < a[i] / 2; j++) {
                sb.append((char) (i + 'A'));
            }

            if (oddCount >= 2) {
                System.out.println("I'm Sorry Hansoo");
                return;
            }
        }

        String result = sb.toString();
        if (oddCount == 1) {
            result += (char) (mid + 'A');
        }
        result += sb.reverse().toString();
        System.out.println(result);
    }
}